Makefile

```diff
@@ -13,146 +12,102 @@ X86_64_ONLY=0
-	PYTHONPATH=src:common ${ENV_DIR}/bin/coverage run -a config_mgr_setup.py test
-	PYTHONPATH=src:common ${ENV_DIR}/bin/coverage run -a plugins_setup.py test
-	${ENV_DIR}/bin/coverage combine
-	${ENV_DIR}/bin/coverage report -m --fail-under 100
-	${ENV_DIR}/bin/coverage xml
-
-.PHONY: env
-env: cluster_mgr_setup.py queue_mgr_setup.py config_mgr_setup.py shared_setup.py plugins_setup.py $(ENV_DIR)/bin/python build-wheelhouse
-
-$(ENV_DIR)/bin/python:
-	# Set up the virtual environment
-	virtualenv --setuptools --python=$(PYTHON_BIN) $(ENV_DIR)
-	$(ENV_DIR)/bin/easy_install "setuptools==24"
-	$(ENV_DIR)/bin/easy_install distribute
-
-BANDIT_EXCLUDE_LIST = src/metaswitch/clearwater/queue_manager/test/,src/metaswitch/clearwater/plugin_tests/,src/metaswitch/clearwater/etcd_tests/,src/metaswitch/clearwater/etcd_shared/test,src/metaswitch/clearwater/config_manager/test/,src/metaswitch/clearwater/cluster_manager/test/,common,_env,.wheelhouse,debian,build_clustermgr,build_configmgr,build_shared
-include build-infra/cw-deb.mk
-include build-infra/python.mk
+run_test: queue_mgr_setup.py config_mgr_setup.py cluster_mgr_setup.py env ${ENV_DIR}/.test_requirements
+	PYTHONPATH=src:common ${PYTHON} cluster_mgr_setup.py test -v && PYTHONPATH=src:common ${PYTHON} queue_mgr_setup.py test -v && PYTHONPATH=src:common ${PYTHON} config_mgr_setup.py test -v && PYTHONPATH=src:common ${PYTHON} plugins_setup.py test -v
 
-.PHONY: build-wheelhouse
 
-define python_component
-# 1 Component Name - dash separated
 
-build-wheelhouse: ${ENV_DIR}/.$1-build-wheelhouse
-
-${ENV_DIR}/.$1-build-wheelhouse: \
-	$$(subst -,_,$1)_setup.py \
-	shared_setup.py \
-	common/setup.py \
-	$(shell find src/metaswitch -type f -not -name "*.pyc") \
-	$(shell find common/metaswitch -type f -not -name "*.pyc") \
-	src/metaswitch/clearwater/$$(subst mgr,manager,$$(subst -,_,$1))/alarm_constants.py \
-	$$(subst -,_,$1)-requirements.txt \
-	common/requirements-test.txt \
-	common/requirements.txt \
-	fv-requirements.txt \
-	shared-requirements.txt
-
-	rm -f $$@
-
-	# Ensure we have an up to date version of pip with wheel support
-	${PIP} install --upgrade pip==9.0.1
-	${PIP} install wheel==0.30.0
-
-	# Generate wheels
-	${PYTHON} $$(subst -,_,$1)_setup.py build -b build_$$(subst -,,$1) bdist_wheel -d $$(subst -,_,$1)_wheelhouse
-	${PYTHON} shared_setup.py build -b build_shared bdist_wheel -d $$(subst -,_,$1)_wheelhouse
-	cd common && WHEELHOUSE=../$$(subst -,_,$1)_wheelhouse make build_common_wheel
+${ENV_DIR}/.test_requirements: common/requirements-test.txt fv-requirements.txt ${ENV_DIR}/.wheels-installed
+	${PIP} install -r common/requirements-test.txt -r fv-requirements.txt
 
-	# Download the required dependencies
-	${PIP} wheel -w $$(subst -,_,$1)_wheelhouse -r $$(subst -,_,$1)-requirements.txt -r shared-requirements.txt -r common/requirements.txt --find-links $$(subst -,_,$1)_wheelhouse
+# Macro to define how the various etcd targets use python common
+define python_common_component
```

Comment:
I think we should name this so it's clear it belongs to etcd, now we have `python_component` (owned by build-infra) and `python_common_component` owned by etcd.